enhancements/update/automatic-updates.md

```diff
@@ -0,0 +1,166 @@
+
+There is also a security risk where a compromised upstream Cincinnati could recommend cluster updates that were not in the cluster's best interest (e.g. 4.2.4 -> 4.1.0).
+This is mitigated by:
+
+* [The configurable `upstream` property][api-upstream].
+    Administrators who do not trust the default `upstream` to be sufficiently reliable may point their clusters at an upstream that they control which they can secure to their satisfaction.
+    The drawback to this approach would be that now they need tooling to approve graph updates as Red Hat makes changes, which will introduce some delays to security-fix rollouts.
+    But that approval would scale per-custom-Cincinnati instead of scaling per-cluster.
+
+There are also potential future mitigations:
+
+* The cluster-version operator could be taught to limit automatic upgrades to those which appear in the [signed metadata][cluster-version-operator-release-metadata] of either the source or target release.
+    The drawback to this approach is that upgrades would be limited to those expected when the releases were created and signed.
+
+## Design Details
+
+### Test Plan
+
+In addition to unit tests, this feature will be tested in a fleet of canary clusters which will run with automatic updates enabled.
+Having the canaries successfully and automatically update in to and out of a candidate release will be part of release and update stabilization criteria.
+
+### Graduation Criteria
+
+The ClusterVersion object is already GA, so there would be nothing to graduate or space to cool a preview implementation.
+
+### Upgrade / Downgrade Strategy
+
+The YAML rendering of the old and updated type are compatible, with the only difference being downgrades to earlier releases (where the property was not available) will clear any stored value.
+Upon upgrading back to a version where the property is available, administrators would need to manually set it to restore automatic upgrade behavior.
+
+### Version Skew Strategy
+
+As a boolean property, this would be [`false` by default][go-zero-values] on upgrades to the updated ClusterVersion Custom Resource Definition.
+This preserves semantics for existing clusters, where manual updates are the only option.
+
+The installer or administrator could set `automaticUpdates` `true` at cluster-creation time if they wished to have it enabled out of the box on new clusters.
+
+## Implementation History
+
+* [API pull request][api-pull-request].
+
+## Drawbacks
+
+The drawbacks are increased exposure to [automatic-update risks](#risks-and-mitigations).
+
+## Alternatives
+
+A [schedule structure][schedule-proposal] like:
+
+```yaml
+automaticUpdates:
+  schedule:
+  - Mon..Fri *-2..10-* 10:00,14:00
+  - Tues..Thurs *-11..1-* 11:00
+```
+
+The schedule structure was designed to support maintenance windows, allowing for updates on certain days or during certain times.
```

Comment:
The entries shown above are punctual datetimes, not maintenance windows.
That is, they specify when a maintenance event can *start* (edge-trigger) but not how long it can be *active* (level-trigger). This is relevant semantic detail if there is any polling logic involved.

Reply:
> That is, they specify when a maintenance event can start (edge-trigger) but not how long it can be active (level-trigger).

There's no cap on upgrade duration.  If you're concerned about it, you'd have small windows early in your day for initiating upgrades, to leave lots of time for monitoring/recovering before you went home.  But this is an alternative proposal; I dunno how deeply we want to dig into it here.

Reply:
Oh sorry, I realize my comment was not worded clearly enough. I'm not trying to cap upgrade duration.

I'm stating that the schedule format would need to define a timespan, even just for the starting event.
That is, "allow auto-update events between 02:00 and 03:00 on Sat" instead of "allow auto-update events exactly at 02:00:00.00 on Sat".

The current syntax is taken from [systemd timestamps](https://www.freedesktop.org/software/systemd/man/systemd.time.html#Displaying%20Timestamps), which defines a single specific point in time instead.

But I guess all of these are implementations details, so feel free to defer this discussion.